src/messaging/tests.py

```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.test import TestCase
+from django.core.urlresolvers import reverse
+from rest_framework.test import APIClient, APITestCase
+
+from django.db.models import Q
+from django.contrib.auth.models import User
+from users.models import Profile
+from .models import Message, Thread
+from users import factories as users_factories
+from .factories import MessageFactory, ThreadFactory
+from .serializers import MessageSerializer, ThreadSerializer
+import random
+
+class SendMessageTest(APITestCase):
+
+    def setUp(self):
+        self.me = users_factories.ProfileFactory()
+        self.client.force_authenticate(user=self.me.user)
+        self.other = users_factories.ProfileFactory()
+        
+
+    def tearDown(self):
+        Message.objects.all().delete()
+        self.me.user.delete()
+        self.other.user.delete()
+
+    def test_send_new_thread(self):
+        create_url = reverse('messaging:send_get_messages', kwargs={'profile_id': self.other.id})
+
+        message_body = 'Test Message'
+        request_params = {
+            'body': message_body,
+        }
+
+        self.assertEqual(len(Thread.objects.all()), 0)
+
+        resp = self.client.post(
+            create_url,
+            data=request_params,
+            )
+
+        new_thread = Thread.objects.get(profile_1=self.me, profile_2=self.other)
```

Comment:
let's update the query here.  This test assumes that the sender will be `profile_1`, which isn't necessarily required.